NotesBackEnd/src/Notes.Contracts/ApiServices/IUrlLocationHelper.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Notes.Contracts.ApiServices
+{
+    public interface IUrlLocationHelper
+    {
+        String GetUrlWithId(string routeName, Guid id);

```

Comment:
The location helper is just a thin wrapper around the `UrlHelper` from WebApi. I think it could take the responsibility for route names, but then you'd need to extract the route name from the controller. So I'm not really sure how good an idea it is.

Reply:
I am not sure if I fulfill criteria right now - I added routeDictionary which contains both route and routeName and controller get value from it.